KB about all AR actions (#135)
telerik/fiddler-everywhere-docs: knowledge-base/using-ar-predefined-actions.md

---
title: Fiddler Auto Responder Predefined Actions
description: Fiddler Everywhere comes with multiple predefined Auto Responder actions that allow you to mock different HTTP responses out of the box.
type: how-to
slug: fiddler-ar-actions
tags: Fiddler Everywhere Auto Responder templates, Auto Responder actions, predefined Fiddler actions
publish: true
res_type: kb
---

#### Environment

|   |   |
|---|---|
| Product  | Fiddler Everywhere |
| Product Version | 1.5.0 and above  |
| Operating System | macOS, Windows, and Linux |

#### Description

The [Auto Responder]({%slug auto-responder-general%}) is one of the most powerful debugging functionalities that Fiddler Everywhere provides. The main idea behind the Auto Responder is to provide a set of rules (based on match rules and actions) that quickly mock a server response and behavior without going through the trouble of modifying the server. Within seconds, you could execute many complex and different rules and test your site against multiple scenarios.

> This article is listing in detail each of the predefined actions listed in the Actions drop-down as DAT files.

Reviewer: ```suggestion
This article lists in detail each of the predefined actions available in the **Actions** drop-down as DAT files. We also explain how to customize the Auto Responder by adding a [Custom DAT file](#custom-dat-file) or a [Custom HTTP Response](#custom-http-response).
```

    Content-Type: text/html
    Content-Length: 520

    Fiddler: HTTP/401 Digest Server Auth Required. 
    ```

1. **403_AuthDeny.dat**

    ```HTTP
    HTTP/1.1 403 Access Denied
    FiddlerTemplate: True
    Date: Fri, 25 Jan 2013 16:49:29 GMT
    Content-Length: 520

    Fiddler: HTTP/403 Access Denied.   
    ```

1. **404_Plain.dat**

    ```HTTP
    HTTP/1.1 404 Not Found
    FiddlerTemplate: True
    Date: Fri, 25 Jan 2013 16:49:29 GMT
    Content-Type: text/html
    Content-Length: 520

    Fiddler: HTTP/404 Not Found 
    ```

1. **407_ProxyAuthBasic.dat**

    ```HTTP
    HTTP/1.1 407 Proxy Auth Required
    FiddlerTemplate: True
    Date: Fri, 25 Jan 2013 16:49:29 GMT
    Proxy-Authenticate: Basic realm="Fiddler (just hit Ok)"
    Content-Type: text/html
    Content-Length: 520

    Fiddler: HTTP/407 Proxy Auth Required.
    ```

1. **502_Unreachable.dat**

    ```HTTP
    HTTP/1.1 502 Unreachable Server
    Date: Fri, 25 Jan 2013 16:49:29 GMT
    FiddlerTemplate: True
    Content-Type: text/html
    Content-Length: 520

    Fiddler: HTTP/502 unreachable server. 
    ```

## Custom DAT File

> The Auto Responder actions drop-down provides an option to load your own custom DAT file with an entirely custom HTTP response. 

Reviewer: ```suggestion
The Auto Responder **Actions** drop-down provides an option to load your own custom DAT file with an entirely custom HTTP response:
```

> - Scroll to the bottom of the drop-down Actions menu and select **Choose saved response file...**.

Reviewer: ```suggestion
1. Scroll to the bottom of the drop-down Actions menu and select **Choose saved response file...**.
```

> - In the file selection window, select your own DAT file and load it as an action rule.

Reviewer: ```suggestion
1. In the file selection window, select your own DAT file and load it as an action rule.
```

![Custom DAT file](../images/livetraffic/autoresponder/kb-auroresponder-custom-dat.png)

## Custom HTTP Response

> Another option that the Auto Responder provides is to directly create the mocked custom HTTP response.

Reviewer: ```suggestion
Another option that the Auto Responder provides is to directly create the mocked custom HTTP response:
```

> - Scroll to the bottom of the drop-down Actions menu and select **Return manually crafted response**.

Reviewer: ```suggestion
1. Scroll to the bottom of the drop-down Actions menu and select **Return manually crafted response**.
```

> - Write the custom response in the **Raw** inspector.

Reviewer: ```suggestion
1. Write the custom response in the **Raw** inspector.
```
